hs_core/management/commands/update_download_counts.py

```diff
@@ -0,0 +1,229 @@
+        )
+        parser.add_argument(
+            '--dry-run',
+            action='store_true',
+            dest='dry_run',
+            help='Run without actually saving changes to the database',
+        )
+        parser.add_argument(
+            '--skip-duplicates',
+            action='store_true',
+            dest='skip_duplicates',
+            help='Skip duplicate resource IDs instead of raising an error',
+        )
+
+    def handle(self, *args, **options):
+        csv_file_path = options['csv_file']
+        dry_run = options['dry_run']
+        skip_duplicates = options['skip_duplicates']
+
+        # Validate CSV file exists
+        if not os.path.exists(csv_file_path):
+            raise CommandError(f"CSV file not found: {csv_file_path}")
+
+        if dry_run:
+            self.stdout.write(
+                self.style.WARNING("DRY RUN MODE - No changes will be saved to the database")
+            )
+
+        # Pre-validate CSV for duplicates
+        self.stdout.write("Validating CSV file for duplicates...")
+        duplicates = self.validate_csv_uniqueness(csv_file_path)
+
+        if duplicates and not skip_duplicates:
+            raise CommandError(
+                f"Found {len(duplicates)} duplicate resource ID(s): {', '.join(sorted(duplicates))}\n"
+                f"Use --skip-duplicates to skip duplicates instead of failing."
+            )
+        elif duplicates and skip_duplicates:
+            self.stdout.write(
+                self.style.WARNING(f"Found {len(duplicates)} duplicate resource ID(s) that will be skipped")
+            )
+
+        success_count = 0
+        error_count = 0
+        skipped_count = 0
+        duplicate_count = 0
+
+        # Track resource IDs to detect duplicates
+        seen_resource_ids = set()
+
+        try:
+            with open(csv_file_path, 'r') as file:
+                # Try different delimiters
+                sniffer = csv.Sniffer()
+                sample = file.read(1024)
+                file.seek(0)
+
+                if sniffer.has_header(sample):
+                    dialect = sniffer.sniff(sample)
+                    reader = csv.reader(file, dialect)
+                    # Skip header row
+                    next(reader, None)
+                else:
+                    # Fallback to default CSV format
+                    file.seek(0)
+                    reader = csv.reader(file)
+
+                # Process each row
+                for row_number, row in enumerate(reader, 1):
+                    if len(row) < 2:
+                        self.stdout.write(
+                            self.style.WARNING(f"Row {row_number}: Skipped - insufficient columns")
+                        )
+                        skipped_count += 1
+                        continue
+
+                    short_id = row[0].strip()
+
+                    # Check for duplicate resource IDs
+                    if short_id in seen_resource_ids:
```

Comment:
Should be in the validation step. Supporting duplicate skipping is making the code more complex than it needs to be.